specifyweb/frontend/js_src/lib/components/Statistics/hooks.tsx

```diff
@@ -250,42 +208,36 @@ export function useResolvedStatSpec(
   }, [item]);
 }
 
-export function setAbsentCategoriesToFetch(
-  layout: StatLayout | undefined,
-  currentCategoriesToFetch: RA<string>,
-  setCategoriesToFetch: (currentCategories: RA<string>) => void
-) {
-  if (layout === undefined) return;
-  const categoriesToFetch: WritableArray<string> = [];
-  layout.forEach((pageLayout) =>
-    pageLayout.categories.forEach(({ items, categoryToFetch }) => {
-      if (items === undefined && categoryToFetch !== undefined) {
-        categoriesToFetch.push(categoryToFetch);
-      }
-      (items ?? []).forEach((item) => {
-        if (
-          item.type === 'DefaultStat' &&
-          item.itemType === 'BackEndStat' &&
-          item.itemValue === undefined &&
-          item.itemName === 'phantomItem'
+export function getDynamicCategoriesToFetch(
+  layout: RA<StatLayout> | undefined
+): RA<string> {
+  return Array.from(
+    new Set(
+      (layout ?? []).flatMap(({ categories }) =>
+        categories.flatMap(({ items }) =>
+          items
```

Comment:
This can be refactored to remove the need for type assertions:

```js
filterArray(
  items.map(item=>
     item.type === 'DefaultStat' && ... /* your condition here */
       ? generateStatUrl( ... /* arguments, without type assertions */)
       : undefined
  )
)
```